docs/src/content/en/docs/agents/guardrails.mdx

```diff
@@ -0,0 +1,333 @@
+        strategy: 'redact',
+        redactionMethod: 'mask',
+        detectionTypes: ['email', 'phone', 'credit-card'],
+        instructions: "Detect and mask personally identifiable information."
+      })
+    ],
+    outputProcessors: [
+      new PIIDetector({
+        // ...
+      })
+    ]
+  });
+  ```
+
+  > See [PIIDetector](../../reference/processors/pii-detector.mdx) for a full list of configuration options.
+  </Tabs.Tab>
+</Tabs>
+
+## Applying multiple processors
+
+You can apply multiple processors by listing them in the `inputProcessors` or `outputProcessors` array. They run in sequence, with each processor receiving the output of the one before it.
+
+A typical order might be:
+
+1. **Normalization**: Standardize input format (`UnicodeNormalizer`).
+2. **Security checks**: Detect threats or sensitive content (`PromptInjectionDetector`, `PIIDetector`).
+3. **Filtering**: Block or transform messages (`ModerationProcessor`).
+
+The order affects behavior, so arrange processors to suit your goals.
+
+```typescript filename="src/mastra/agents/test-agent.ts" showLineNumbers copy
+import {
+  UnicodeNormalizer,
+  ModerationProcessor,
+  PromptInjectionDetector,
+  PIIDetector
+  } from "@mastra/core/processors";
+
+export const testAgent = new Agent({
+  // ...
+  inputProcessors: [
+    new UnicodeNormalizer({
+      //...
+    }),
+    new PromptInjectionDetector({
+      // ...
+    }),
+    new PIIDetector({
+      // ...
+    }),
+    new ModerationProcessor({
+      // ...
+    })
+  ],
+});
+```
+
+## Understanding responses
+
+Processors behave differently depending on their purpose:
+
+- **Transformers**: Modify messages (e.g. normalize input, translate language). These allow processing to continue.
+- **Validators**: Check for violations (e.g. prompt injection, PII, harmful content) and may stop further processing if certain conditions are not met.
+
+If a validator stops processing:
+
+- No subsequent processors in the same `inputProcessors` or `outputProcessors` array are run.
+- The agent still returns a 200 response.
+
+In all cases:
+
+- The response includes details about what was detected and how it was handled.
+
+### PIIDetector example
+
+For example, if an agent uses the `PIIDetector` and a `.generate()` or `.stream()` call includes a credit card number:
+
+```typescript
+const result = await agent.generate("Is this credit card number correct? 4543 1374 5089 4332");
+
+console.log(result.text);
+
+const stream = await agent.stream("Is this credit card number correct? 4543 1374 5089 4332");
+
+for await (const chunk of stream.textStream) {
+  process.stdout.write(chunk);
+}
+```
+
+The response would include a message from the agent along with details explaining what was detected and how it was handled.
+
+```text
+[PIIDetector] Redacted PII: PII detected. Types: credit-card. Detections: 1 items
+
+I'm sorry, but I can't assist with entering or processing sensitive information like credit card numbers.
+```
```

Comment:
@akuya-ekorot FYI

Reply:
@DanielSLew, does this sound ok? 

i've tried to group the _types_ of processors to explain that some have pass/fail checks, while others "pass through" like `LanguageDetector`. 

E.g, `LanguageDetector` would never abort would it? Any processor added after a `LanguageDetector` in an array would still run regardless. 

Where as, if a `PIIDetector` aborts, subsequent processors wouldn't run?